library/std/src/sys/unix/mod.rs

```diff
@@ -86,6 +93,56 @@ pub fn init() {
         reset_sigpipe();
     }
 
+    // In the case when all file descriptors are open, the poll has been
+    // observed to perform better than fcntl (on GNU/Linux).
+    #[cfg(not(any(
+        target_os = "emscripten",
+        target_os = "fuchsia",
+        // The poll on Darwin doesn't set POLLNVAL for closed fds.
+        target_os = "macos",
+        target_os = "ios",
+        target_os = "redox",
+    )))]
+    unsafe fn sanitize_standard_fds() {
+        use crate::sys::os::errno;
+        let pfds: &mut [_] = &mut [
+            libc::pollfd { fd: 0, events: 0, revents: 0 },
+            libc::pollfd { fd: 1, events: 0, revents: 0 },
+            libc::pollfd { fd: 2, events: 0, revents: 0 },
+        ];
+        while libc::poll(pfds.as_mut_ptr(), 3, 0) == -1 {
```

Comment:
Oh, this will be "interesting" to support on Miri...

Any chance we could add `cfg(miri)` to the list of exceptions above?

Reply:
Which implementation would you like to use: `fcntl(fd, F_GETFD)` and `open("/dev/null")`, or empty one?

Reply:
The empty on would be best. The standard FDs will always be available in Miri (and they cannot even be closed by the application as the interpreter cannot implement that behavior due to https://github.com/rust-lang/rust/issues/40032).

Reply:
Miri should use empty implementation now.